src/main/java/hudson/tasks/junit/ClassResult.java

```diff
@@ -188,10 +190,13 @@ else if(r.isPassed()) {
             else {
                 failCount++;
             }
-            //retrieve the class duration from these cases' suite time
-            if(duration == 0){
-                duration = r.getSuiteResult().getDuration();
-            }
+            suites.add( r.getSuiteResult() );
+        }
+
+        // retrieve the class duration from these cases' suite time
+        duration = 0;
+        for (SuiteResult s : suites) {
+            duration += s.getDuration();
```

Comment:
i'm afraid not...and a change like that would be caught by the one of my new tests
in case not...i should add a case for this too ;)

..the for cycle is running thru cases...and the old trick was to only use the case runtime when the total is 0...the new code collects all the unique suites and aggregates there sum time